### DISCUSSION TITLE
IConfirmNavigationAsync.CanNavigateAsync not called from iOS "Back" Button

### DISCUSSION THREAD

**Discussion by munkii** [PrismLibrary/Prism#3285] [.NET MAUI]
I have implemented IConfirmNavigationAsync.CanNavigateAsync on a ViewModel. 

When the user presses the Back arrow in the Navbar in Android CanNavigateAsync  is called and I can stop the Navigation if needed,

When the user presses the "Back" button in the Navigation Bar in iOS CanNavigateAsync is not called. I assume this should just work without me having to do anything particular.

The View\ViewModel is not being loaded modally and is simply using the INavigationService. What should I be doing?

```
var navigationParams = new NavigationParameters();

if (data != null)
{
    navigationParams.Add("initData", data);
}

if (animate == true)
{
    navigationParams.Add(KnownNavigationParameters.Animated, true);
}

if (modal == true)
{
    navigationParams.Add(KnownNavigationParameters.UseModalNavigation, true);
}

await this.NavigationService.NavigateAsync(typeof(T).Name, navigationParams);
```

**Comment by christian-strydom** [comment]
I'm facing a similar issue, working on Android as expected, but iOS's back button isn't firing the CanNavigateAsync method. Interestingly enough when I call navigationService.GoBackAsync(); in my ViewModel it does hit. Have you managed to find the cause for the problem why the iOS title view back button is not hitting CanNavigateAsync? 

Initially I thought it could be caused due to my custom `<NavigationPage.TitleView>`, I changed it to `<prism:PrismNavigationPage.TitleView>` but unfortunately it is still not working on iOS.

**Comment by munkii** [comment]
I have not managed to find the problem but is almost certainly related to this discussion I opened in May 2024, [Correct use of NavigationPage](https://github.com/PrismLibrary/Prism/discussions/3142)

**Comment by christian-strydom** [comment]
I've created an issue [here](https://github.com/PrismLibrary/Prism/issues/3311)

**Reply by munkii** [reply]
Thanks
